datafusion/sql/src/expr/mod.rs

```diff
@@ -165,7 +165,9 @@ impl<S: ContextProvider> SqlToRel<'_, S> {
         schema: &DFSchema,
         planner_context: &mut PlannerContext,
     ) -> Result<Expr> {
-        let mut expr = self.sql_expr_to_logical_expr(sql, schema, planner_context)?;
+        // The location of the original SQL expression in the source code
+        let mut expr =
+            self.sql_expr_to_logical_expr(sql.clone(), schema, planner_context)?;
```

Comment:
We can now remove this clone, right? I tried it locally and it worked fine

Reply:
It works! Thank you :)